tests/messages/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+#![cfg_attr(not(feature = "std"), feature(alloc_error_handler))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+#[cfg(not(feature = "std"))]
+use gstd::{prelude::*, *};
+
+#[cfg(feature = "std")]
+#[cfg(test)]
+mod native {
+    include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+}
+
+#[derive(Encode, Debug, Decode, PartialEq)]
+pub enum Request {
+    SendOnce,
+    SendInf,
+    SendPushAfterCommit,
+    ReplyOnce,
+    ReplyTwice,
+    ReplyPushAfterReply,
+}
+
+#[derive(Encode, Debug, Decode, PartialEq)]
+pub enum Reply {
+    Empty,
+    Error,
+}
+
+#[cfg(not(feature = "std"))]
+mod wasm {
+    extern crate alloc;
+
+    use codec::{Decode, Encode};
+    use gstd::{ext, msg, prelude::*, MessageId, ProgramId};
+
+    use super::{Reply, Request};
+
+    mod gear_sys {
+        extern "C" {
+            pub fn gr_send_init() -> u32;
+            pub fn gr_send_push(handle: u32, data_ptr: *const u8, data_len: u32);
+            pub fn gr_send_commit(
+                handle: u32,
+                message_id_ptr: *mut u8,
+                program: *const u8,
+                gas_limit: u64,
+                value_ptr: *const u8,
+            );
+        }
+    }
```

Comment:
Isn't it better to use `gcore` here? Why redeclare syscalls?

Reply:
`gcore::msg::sys` is private. This PR also makes sure that `gcore` message handle is not reusable and is consumed after committing a message, so `gcore` can't be used for that particular test

Reply:
`gcore::msg` has the same `send_init`. `send_push`, and `send_commit`.

Reply:
`gcore::msg::send_push` and `gcore::msg::send_push` require `gcore::MessageHandle`, that gets consumed in `gcore::msg::send_commit`. This PR also makes sure that `gcore::MessageHandle` does not implement `Clone`. Unless I'm missing something, it should be impossible to recreate `send_push_after_commit` test without using syscalls. Test `send_push_after_commit` validates behavior of the runner, rather than `gcore`/`gstd` libraries, making sure that correct trap gets raised during execution.